mkosi/__init__.py

```diff
@@ -2625,6 +2640,16 @@ def install_debian_or_ubuntu(args: MkosiArgs, root: Path, *, do_run_build_script
 
     invoke_apt(args, do_run_build_script, root, "install", ["--no-install-recommends", *extra_packages])
 
+    # Now clean up and add the real repositories, so that the image is ready
+    if args.use_mirror_verbatim:
+        if args.distribution == Distribution.ubuntu:
+            main_repo = f"deb http://archive.ubuntu.com/ubuntu {args.release} {' '.join(repos)}\n"
+        else:
+            main_repo = f"deb http://deb.debian.org/debian {args.release} {' '.join(repos)}\n"
+        root.joinpath("etc/apt/sources.list").write_text(main_repo)
+        root.joinpath("etc/apt/sources.list.d/mirror.list").unlink()
+        add_apt_auxiliary_repos(args, root, repos)
```

Comment:
Technically the mirrors are only intended for building the image, not necessarily to make sure installing packages afterwards works correctly, it just happens to work that way for apt because we have to run it from inside the container unlike the other package managers. But I guess there's no reason to regress here unnecessarily so we can keep this.